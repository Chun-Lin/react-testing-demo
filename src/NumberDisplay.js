import React from 'react'

const NumberDisplay = ({number}) => {
  return (
    <div>
      <span data-testid="number-display">{number}</span>
    </div>
  )
}

export default NumberDisplay