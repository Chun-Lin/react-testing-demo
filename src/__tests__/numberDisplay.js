import React from 'react'
import { render, cleanup } from '@testing-library/react'
import NumberDisplay from 'NumberDisplay'

afterEach(cleanup)

test('calling render with the same component on the same container does not remount', () => {
  const { getByTestId, rerender } = render(<NumberDisplay number={1} />)
  expect(getByTestId('number-display').textContent).toBe('1')

  // re-render the same component with different props
  rerender(<NumberDisplay number={2} />)
  expect(getByTestId('number-display').textContent).toBe('2')
})

test('<NumberDisplay> snapshot', () => {
  const { container } = render(<NumberDisplay number={1} />)
  expect(container).toMatchSnapshot()
})