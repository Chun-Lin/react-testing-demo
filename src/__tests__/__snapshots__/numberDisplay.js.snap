// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberDisplay> snapshot 1`] = `
<div>
  <div>
    <span
      data-testid="number-display"
    >
      1
    </span>
  </div>
</div>
`;
